### ISSUE TITLE
impl TextLen for &'_ Arc<String> 

### ISSUE THREAD

**Issue by Matthias-Fauconneau** [rust-analyzer/text-size#45]
Is this intentionally missing ?
Would be nice :)
impl TextLen for &'_ Arc<String> { fn text_len(self) -> TextSize { self.as_str().text_len() } }

**Comment by matklad** [comment]
Kind of: we just follow the concervative set of impls of std::Pattern: https://doc.rust-lang.org/stable/std/str/pattern/trait.Pattern.html#implementors

**Comment by Matthias-Fauconneau** [comment]
Maybe that's a restriction which makes more sense for std though.
But more importantly, this cannot be worked around because of the Sealed :(

**Comment by matklad** [comment]
Yes, this needs to be fixed at the call-site by calling `.as_str()`, which doesn't seem like the end of the world to me :)

**Comment by CAD97** [comment]
That's definitely out-of-date documentation on `TextSize::of`; I wrote it when we had much more aggressive blanket impls for `TextLen`.

Our general consensus (for 1.0 at least) was that `TextSize::of` should be used just for primitives (i.e. what was `of_str` and `of_char` in `text_unit`), and compound types could either provide a primitive conversion (to feed into `TextSize::of`), or its own ad-hoc accessor.

**Comment by CAD97** [comment]
> `we just follow the concervative set of impls of std::Pattern`

We should probably add an impl for `&[char]` then...
